### PR TITLE
Escape route values for AutoClient

### PR DESCRIPTION
**Pull request by sebastienros**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4423)

### CODE REVIEW

**Review by geeknoid — APPROVED** [review]

File: src/Generators/Microsoft.Gen.AutoClient/Emitter.cs [hunk 1]
```diff
@@ -226,7 +226,22 @@ private void GenRestApiMethod(RestApiMethod restApiMethod, RestApiType restApiTy
...
+        var encodedPath = restApiMethod.Path;
+
+        if (encodedPath != null && restApiMethod.FormatParameters.Count > 0)
+        {
+            foreach (var param in restApiMethod.FormatParameters)
+            {
+                var escapedParamName = $"{param}Escaped";
+                OutLn($"var {escapedParamName} = {Uri}.EscapeDataString($\"{{{param}}}\");");
+
+                encodedPath = encodedPath.Replace($"{{{param}}}", $"{{{escapedParamName}}}");
```
Comment (xakep139) [inline src/Generators/Microsoft.Gen.AutoClient/Emitter.cs:238]: Can we do the replacement on the `StringBuilder`? This will give us a better performance

**Comment by RussKie** [comment]
Is this required for .NET 8?

File: src/Generators/Microsoft.Gen.AutoClient/Emitter.cs (hunk 1 above)
Comment (sebastienros) [reply src/Generators/Microsoft.Gen.AutoClient/Emitter.cs]: Updated

**Pull request merged**